app/com/gu/memsub/auth/common/MemSub.scala

```diff
@@ -37,12 +32,18 @@ object MemSub {
     }
 
     def googleGroupCheckerFor(config: Config): GoogleGroupChecker = {
-      val con = config.getConfig("google.directory.service_account")
-      new GoogleGroupChecker(con.getString("email"), ServiceAccountCredentials
-        .newBuilder()
-        .setClientEmail(con.getString("id"))
-        .setPrivateKey(ServiceAccount.PrivateKey)
-        .build())
+      val request = new GetObjectRequest("membership-private", "google-auth-service-account-certificate.json")
+      AwsS3.fetchObject(AwsS3.client, request).map { stream =>
+        val googleServiceAccountCredential = ServiceAccountCredentials.fromStream(stream)
+        stream.close()
+
+        val impersonatedUser = config.getString("google.oauth.impersonatedUser")
+
+        new GoogleGroupChecker(
+          impersonatedUser,
+          googleServiceAccountCredential
+        )
+      }.get // created on startup
```

Comment:
This will throw if there's an issue with S3 or the config. The `GoogleGroupChecker` is required on startup so I think this is ok